indy_node/server/action_req_handler.py

```diff
@@ -0,0 +1,94 @@
+
+    def __init__(self, idrCache: IdrCache,
+                 restarter: Restarter, poolManager, poolCfg: PoolConfig):
+        self.idrCache = idrCache
+        self.restarter = restarter
+        self.poolManager = poolManager
+        self.poolCfg = poolCfg
+
+    def doStaticValidation(self, request: Request):
+        identifier, req_id, operation = request.identifier, request.reqId, request.operation
+        if operation[TXN_TYPE] == POOL_RESTART:
+            self._doStaticValidationPoolRestart(identifier, req_id, operation)
+
+    def _doStaticValidationPoolRestart(self, identifier, req_id, operation):
+        if SCHEDULE in operation.keys() is None and operation[SCHEDULE] != "0":
+            try:
+                dateutil.parser.parse(operation[SCHEDULE])
+            except Exception:
+                raise InvalidClientRequest(identifier, req_id,
+                                           "time is not valid")
+
+    def validate(self, req: Request):
+        status = None
+        operation = req.operation
+        typ = operation.get(TXN_TYPE)
+        if typ not in [POOL_RESTART]:
+            return
+        origin = req.identifier
+        try:
+            origin_role = self.idrCache.getRole(origin, isCommitted=False)
+        except BaseException:
+            raise UnauthorizedClientRequest(
+                req.identifier,
+                req.reqId,
+                "Nym {} not added to the ledger yet".format(origin))
+        action = ""
+        if typ == POOL_RESTART:
+            action = operation.get(ACTION)
+        r, msg = Authoriser.authorised(
+            typ, origin_role, field=ACTION, oldVal=status, newVal=action)
+        if not r:
+            raise UnauthorizedClientRequest(
+                req.identifier, req.reqId, "{} cannot do restart".format(
+                    Roles.nameFromValue(origin_role)))
+
+    def apply(self, req: Request, cons_time: int = None):
+        if req.txn_type != POOL_RESTART:
+            raise InvalidClientRequest("{} is not type of action transaction"
+                                       .format(req.txn_type))
+        result = {}
+        try:
+            txn = reqToTxn(req)
+            self.restarter.handleActionTxn(txn)
+            result = self._generate_action_result(req)
+        except Exception as ex:
+            result = self._generate_action_result(req,
+                                                 False,
+                                                 ex.args[0])
+            logger.warning("Restart is failed")
+        finally:
+            return result
+
+    def _generate_action_result(self, request: Request, is_success=True,
+                                msg=None):
+        return {TXN_TYPE: request.operation.get(TXN_TYPE),
+                      f.IDENTIFIER.nm: request.identifier,
+                      f.REQ_ID.nm: request.reqId,
+                      f.IS_SUCCESS.nm: is_success,
+                      f.MSG.nm: msg}
```

Comment:
Action result must also contain the request-specific fields. So return the merged dictionary:
`
        return {**request.operation, **{
            f.IDENTIFIER.nm: request.identifier,
            f.REQ_ID.nm: request.reqId,
            f.IS_SUCCESS.nm: is_success,
            f.MSG.nm: msg}
        }}
`
So `type` and request-specific fields will be taken from `request.operation`.